### PR TITLE
[ BLAS ] Implement explicit scopy function for AVX2

### PR DESCRIPTION
**Pull request by skykongkong8**
- Occasional segmentation fault when scopy_cblas detected from mixed precision model test.
- Implement hand-written scopy function with avx2

**Self evaluation:**
1. Build test:     [X]Passed [ ]Failed [ ]Skipped
2. Run test:     [X]Passed [ ]Failed [ ]Skipped

Signed-off-by:skykongkong8 <ss.kong@samsung.com>

### CODE REVIEW

File: nntrainer/tensor/blas_avx.cpp [hunk 1]
```diff
@@ -274,4 +274,40 @@ bool isValid(const size_t N, const float *input) {
...
+    input += 8;
+
+    _mm256_storeu_ps(out_data, (vec));
+    out_data += 8;
+  }
+
+  while (idx < N) {
+    *out_data = *input;
+    ++out_data;
+    ++input;
```
Comment (dkjung) [inline nntrainer/tensor/blas_avx.cpp:308]: Doesn't the following line have the same effect?

```c
*out_data++ = *input++;
```

File: nntrainer/tensor/blas_avx.cpp [hunk 2]
```diff
@@ -274,4 +274,40 @@ bool isValid(const size_t N, const float *input) {
...
+  assert(output != NULL);
+
+  unsigned int idx = 0;
+  unsigned int N16 = (N >> 4) << 4;
+  unsigned int N8 = (N >> 3) << 3;
+  float *out_data = (float *)output;
+
+  for (; idx < N16; idx += 16) {
+    __m256 vec0 = _mm256_loadu_ps(input);
+    __m256 vec1 = _mm256_loadu_ps(input + 8);
```
Comment (dkjung) [inline nntrainer/tensor/blas_avx.cpp:290]: Is stride of 8 correct for `input` and `out_data`(not 16)?

**Review by dkjung — APPROVED** [review]
LGTM, just left 2 minor comments.

File: nntrainer/tensor/blas_avx.cpp (hunk 1 above)
Comment (skykongkong8) [reply nntrainer/tensor/blas_avx.cpp:308]: Good point!

File: nntrainer/tensor/blas_avx.cpp (hunk 2 above)
Comment (skykongkong8) [reply nntrainer/tensor/blas_avx.cpp:290]: As far as I am concerned, won't it be 8 ?

256 = 32*8

1. load 8
2. load next 8
=> 1 + 2 is 16

**Pull request closed without merging**